dashboard/test/integration/charset_test.rb

```diff
@@ -0,0 +1,70 @@
+require 'test_helper'
+
+class CharsetTest < ActionDispatch::IntegrationTest
+  def no_database
+    Rails.logger.info '--------------'
+    Rails.logger.info 'DISCONNECTING DATABASE'
+    Rails.logger.info '--------------'
+
+    ActiveRecord::Base.connection.disconnect!
+  end
+
+  test "attempting to log in as user with utf8mb4 chars does not hit the db" do
+    # make sure all the classes are loaded
+    post '/users/sign_in', login: 'not a user', password: 'not a password'
+    assert_response :success
+    assert_select 'div.alert', 'Invalid email or password.'
+
+    no_database
+
+    panda_panda = "Panda \u{1F43C}"
```

Comment:
Ugh, I have no idea why these tests are failing, the integration tests are really hard to debug in an environment that I dot have access to because the server errors are buried in the logs.

> On Dec 8, 2015, at 14:56, philbogle <notifications@github.com> wrote:
> 
> In dashboard/test/integration/charset_test.rb:
> 
> > +    Rails.logger.info '--------------'
> > +    Rails.logger.info 'DISCONNECTING DATABASE'
> > +    Rails.logger.info '--------------'
> > +
> > +    ActiveRecord::Base.connection.disconnect!
> > +  end
> > +
> > +  test "attempting to log in as user with utf8mb4 chars does not hit the db" do
> > +    # make sure all the classes are loaded
> > +    post '/users/sign_in', login: 'not a user', password: 'not a password'
> > +    assert_response :success
> > +    assert_select 'div.alert', 'Invalid email or password.'
> > +
> > +    no_database
> > +
> > +    panda_panda = "Panda \u{1F43C}"
> Please add a comment that this is a Unicode character whose UTF-8 encoding takes 4 bytes.
> 
> (Most will not know what utf8mb4 means)
> 
> —
> Reply to this email directly or view it on GitHub.
>